### ISSUE TITLE
03-28 Performance regression on YCSB after txn-local-latches was removed

### ISSUE THREAD

**Issue by mahjonp** [pingcap/tidb#16068]
## Bug Report

https://github.com/pingcap/tidb/pull/15765

![image](https://user-images.githubusercontent.com/3251373/78363193-eb15e100-75ed-11ea-9402-92dd390f149f.png)

```
Version:
tidb: c8fa259b2e26eebc274a6637f1ee5dcc13102177(master) 2020-03-28 02:16:33
tikv: ea0fd4d7688d1d462a94cfe4f7ab4d7cb0b30ab5(master) 2020-03-28 08:33:22
pd: a63cf42fe9687c0e580777a01facc4004da69b63(master) 2020-03-28 05:44:57

Compare:
tidb: https://github.com/pingcap/tidb/compare/78ac219...c8fa259
tikv: https://github.com/tikv/tikv/compare/8582315...ea0fd4d
pd: https://github.com/pingcap/pd/compare/9c709cf...a63cf42

type: mysql

workload: <load>

insert: *QPS: 20436.77 ± 0.0727% (std=10.92) delta: 13.88%

workload: <workloada>

update: *QPS: 2590.40 ± 0.6138% (std=10.59) delta: -73.64%
read: *QPS: 2687.80 ± 0.6474% (std=11.81) delta: -74.16%

workload: <workloadb>

update: *QPS: 1716.07 ± 0.2603% (std=3.26) delta: -25.08%
read: *QPS: 33614.28 ± 0.6788% (std=148.17) delta: -23.37%

workload: <workloadc>

read: *QPS: 53017.88 ± 1.1956% (std=388.17) delta: 3.11%

workload: <workloadd>

insert: *QPS: 2397.54 ± 0.9201% (std=13.68) delta: 4.09%
read: *QPS: 45570.26 ± 0.9547% (std=268.01) delta: 3.73%

workload: <workloade>

insert: *QPS: 140.60 ± 9.1750% (std=8.19) delta: 11.71%
scan: *QPS: 2670.50 ± 8.8148% (std=148.38) delta: 11.80%

workload: <workloadf>

read: *QPS: 5172.27 ± 0.2016% (std=6.54) delta: -70.64%
update: *QPS: 2492.45 ± 0.2508% (std=3.78) delta: -70.19%
```

**Comment by siddontang** [comment]
ping @bobotu

**Comment by bobotu** [comment]
The `workloada` has many write conflicts, so it is recommended to run with pessimistic txn mode.
Besides the txn mode, we also need to know that there are differences between the two results.

* The `txn-local-latches` will abort conflicted transactions as soon as possible
* The `go-ycsb` will not retry transactions when meets conflict errors
* The main benchmark loop relies on a random number generator, so the next statement will try to do a read operation or update with different values

In summary, the old benchmark results are better than the new results are in line with expectations.
